### DISCUSSION TITLE
How to get the propeller aerodynamic coefficient ?

### DISCUSSION THREAD

**Discussion by Dobid** [JSBSim-Team/jsbsim#850] [Support requests]
Hello !

As explained in #843, I'm trying to implement an autopilot for the [x8 UAV](http://skywalkermodel.com/en/76.html#inspire-1-pro-overview-s2) based on a PID controller by following this book : Small Unmanned Aircrafts by Beard and McLain (here's the [fdm](https://github.com/AOS55/Fixedwing-Airsim/blob/master/x8/x8.xml#enroll-beta)).

In the book, in the longitudinal autopilot part, they give a set of formulas to compute PID gains for the airspeed controller : 
$k_{iv} = {\omega_{nv}}^2/a_{v2}$ ; $k_{pv} = (2\xi_{v}\omega_{nv} - a_{v1})/a_{v2}$
Those formulas show the following coefficients : $a_{v1}, a_{v2}$. 

Those are transfer function coefficients, their formulas are : 
![image](https://user-images.githubusercontent.com/36502022/222444421-048fd295-baab-448e-9ec5-b1cbabf22877.png)
With $S_{prop}$ being the area swept by the propeller and $C_{prop}$ the aerodynamical coefficient of the propeller, for more context, they appear here, when finding the formula of the thrust produced by the propeller/engine : 
![image](https://user-images.githubusercontent.com/36502022/222444982-902ae496-127b-4ae9-a2ea-de4833a8faa8.png)

So in order to achieve my PID, I need to find a way to get or compute this value of $C_{prop}$ and I have no idea of where to look for. I checked the fdm ([propulsion tag](https://github.com/AOS55/Fixedwing-Airsim/blob/c5054486fb05dc67668b123c7789e399c8d4ce02/x8/x8.xml#L78)) referencing the [engine's xml file](https://github.com/AOS55/Fixedwing-Airsim/blob/master/x8/electric800W.xml) and [the direct thrust file](https://github.com/JSBSim-Team/jsbsim/blob/master/engine/direct.xml).

Can someone help me find a way to compute this $C_{prop}$ value ? The x8 UAV [description table](http://skywalkermodel.com/en/76.html#inspire-1-pro-overview-s3) gives the dimensions of the propeller, but searching the internet about a corresponding $C_{prop}$ doesn't bring anything. This $C_{prop}$ coefficient notation seems to be specific to the book I'm following and I can't link it to information I find online (UAV specs, propeller specs, fdm files...)

Thanks in advance !

**Comment by seanmcleod** [comment]
I don't have a definite answer, but take a look at the following code:

https://github.com/JSBSim-Team/jsbsim/blob/ebb770ebc0ec4cee6aefef6ad88702ef1164b5d7/src/models/propulsion/FGPropeller.cpp#L207-L298

I wonder if, without having a proper look whether $C_{prop}$ is the `thrust_coefficient`.

https://github.com/JSBSim-Team/jsbsim/blob/ebb770ebc0ec4cee6aefef6ad88702ef1164b5d7/src/models/propulsion/FGPropeller.cpp#L154-L155

https://github.com/JSBSim-Team/jsbsim/blob/ebb770ebc0ec4cee6aefef6ad88702ef1164b5d7/src/models/propulsion/FGPropeller.cpp#L118-L120

https://github.com/JSBSim-Team/jsbsim/blob/ebb770ebc0ec4cee6aefef6ad88702ef1164b5d7/src/models/propulsion/FGPropeller.cpp#L103-L104

**Reply by Dobid** [reply]
Thanks for the quick answer, I will go through the code and identify the laws used with the book mentionned in the comments to try and find some similarities with my $C_{prop}$ coefficient. I got assigned some other work before that, I'll keep this thread updated when I find something, or if I have another question.
